packages/preview-middleware-client/src/cpe/changes/validator.ts

```diff
@@ -0,0 +1,33 @@
+import ResourceBundle from 'sap/base/i18n/ResourceBundle';
+
+/**
+ * Function to validate if a given value is a valid binding model.
+ *
+ * @param value value to be checked.
+ */
+export function validateBindingModel(value: string): void {
+    const bindingValue = value.replace(/[{}]/gi, '');
+    const bindingParts = bindingValue
+        .trim()
+        .split('>')
+        .filter((el) => el != '');
+
+    if (!bindingParts.length) {
+        throw new SyntaxError('Invalid binding string.');
+    }
+
+    if (bindingParts[0].trim() === 'i18n') {
+        if (bindingParts.length === 2) {
+            const resourceBundle = ResourceBundle.create({
+                url: '../i18n/i18n.properties'
+            }) as ResourceBundle;
+            if (!resourceBundle.hasText(bindingParts[1].trim())) {
```

Comment:
This `bindingParts[1].trim()` could be extracted in its own variable like **resourceKey**.

Reply:
Done